src/main/java/org/zalando/nakadi/filters/ConsumerHeaderFilter.java

```diff
@@ -0,0 +1,43 @@
+package org.zalando.nakadi.filters;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.annotation.Nullable;
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ConsumerHeaderFilter extends OncePerRequestFilter {
+    public class Subject {
+        public String eid;
+        @Nullable
+        public String xConsumer;
+
+        public Subject(final String eid, final String xConsumer) {
+            this.eid = eid;
+            this.xConsumer = xConsumer;
+        }
+    }
+
+    @Override
+    protected void doFilterInternal(final HttpServletRequest request, final HttpServletResponse response,
+                                    final FilterChain filterChain) {
+
+        final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication instanceof OAuth2Authentication) {
+            final OAuth2Authentication oAuth2Authentication = (OAuth2Authentication) authentication;
+            final String eid = (String) authentication.getPrincipal();
+            final UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(
+                    new Subject(eid, request.getHeader("X-Consumer")), null);
```

Comment:
Do we need this in the open source project?

Reply:
You're right. I am moving this to the plugin. I will update the pull request.